WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java

```diff
@@ -701,268 +681,44 @@ private void updatePostSettingsAndSaveButton() {
-        }
-
-        @Override
-        protected void onPostExecute(Address address) {
-            setLocationStatus(LocationStatus.FOUND);
-            showLocationView();
-
-            if (address != null) {
-                double[] coordinates = GeocoderUtils.getCoordsFromAddress(address);
-                setLocation(coordinates[0], coordinates[1]);
-
-                String locationName = GeocoderUtils.getLocationNameFromAddress(address);
-                updateLocationText(locationName);
-            } else {
-                showLocationNotAvailableError();
-                showLocationSearch();
+            double latitude = args[0];
+            double longitude = args[1];
+            try {
+                return GeocoderUtils.getAddressFromCoords(getActivity(), latitude, longitude);
+            } catch (IllegalArgumentException iae) {
+                return null;
             }
         }
-    }
 
-    private LocationHelper.LocationResult locationResult = new LocationHelper.LocationResult() {
-        @Override
-        public void gotLocation(final Location location) {
-            if (getActivity() == null)
+        protected void onPostExecute(@Nullable Address address) {
+            if (address == null) {
                 return;
-            // note that location will be null when requesting location fails
-            getActivity().runOnUiThread(new Runnable() {
-                public void run() {
-                    setLocation(location);
+            }
+            if (address.getMaxAddressLineIndex() > 0) {
+                StringBuilder sb = new StringBuilder();
+                for (int i = 0; ; ++i) {
+                    sb.append(address.getAddressLine(i));
+                    if (i == address.getMaxAddressLineIndex()) {
+                        sb.append(".");
+                        break;
+                    } else {
+                        sb.append(", ");
+                    }
                 }
-            });
-        }
-    };
-
-    private View mLocationAddSection;
-    private View mLocationSearchSection;
-    private View mLocationViewSection;
-    private TextView mLocationText;
-    private EditText mLocationEditText;
-    private Button mButtonSearchLocation;
-
-    private TextWatcher mLocationEditTextWatcher = new TextWatcher() {
-        @Override
-        public void beforeTextChanged(CharSequence s, int start, int count, int after) { }
-
-        @Override
-        public void onTextChanged(CharSequence s, int start, int before, int count) { }
-
-        @Override
-        public void afterTextChanged(Editable s) {
-            String buttonText;
-            if (s.length() > 0) {
-                buttonText = getResources().getString(R.string.post_settings_search_location);
-            } else {
-                buttonText = getResources().getString(R.string.post_settings_search_current_location);
+                mLocationTextView.setText(sb.toString());
```

Comment:
Added a `isAdded()` check in 5c9953d182 (note: updating views when the fragment is not added is not an issue)